packages/theme_tailor/lib/src/generator/theme_tailor_generator.dart

```diff
@@ -124,9 +140,26 @@ class ThemeTailorGenerator extends GeneratorForAnnotation<Tailor> {
   }
 
   List<String> _computeThemes(ConstantReader annotation) {
-    return List<String>.from(
-      annotation.read('themes').listValue.map((e) => e.toStringValue()),
-    );
+    List<String> annotationThemes;
+    if (!annotation.read('themes').isNull) {
+      annotationThemes = List<String>.from(
+        annotation.read('themes').listValue.map((e) => e.toStringValue()),
+      );
+    } else {
+      annotationThemes = [];
+    }
+
+    var pubspecThemes = (builderOptions.config['themes'] as List)
+        .map((element) => element.toString())
+        .toList();
+
+    if (annotationThemes.isNotEmpty) {
+      return annotationThemes;
+    } else if (pubspecThemes.isNotEmpty) {
+      return pubspecThemes;
+    } else {
+      return ['light', 'dark'];
+    }
```

Comment:
Maybe smth like that. You can for sure make it more pretty

```suggestion
    if (!annotation.read('themes').isNull) {
      return List<String>.from(
        annotation.read('themes').listValue.map((e) => e.toStringValue()),
      );
    }

    var pubThemes = (builderOptions.config['themes'] as List)
        .map((element) => element.toString())
        .toList();
        
    const defaultThemes = ['light', 'dark'];
    
    return pubThemes.isNotEmpty? pubThemes : defaultThemes;
```